packages/selfcare-client-users-updater-consumer/src/services/selfcareClientUsersUpdaterProcessor.ts

```diff
@@ -0,0 +1,118 @@
+
+// eslint-disable-next-line @typescript-eslint/explicit-function-return-type
+export function selfcareClientUsersUpdaterProcessorBuilder(
+  refreshableToken: RefreshableInteropToken,
+  authorizationProcessClient: AuthorizationProcessClient,
+  productId: string,
+  allowedOriginsUuid: string[]
+) {
+  return {
+    async processMessage({
+      message,
+      partition,
+    }: EachMessagePayload): Promise<void> {
+      const correlationId: CorrelationId = generateId();
+
+      const loggerInstance = logger({
+        serviceName: "selfcare-client-users-updater-consumer",
+        correlationId,
+      });
+
+      try {
+        loggerInstance.info(
+          `Consuming message for partition ${partition} with offset ${message.offset}`
+        );
+
+        if (!message.value) {
+          loggerInstance.warn(
+            `Empty message for partition ${partition} with offset ${message.offset}`
+          );
+          return;
+        }
+
+        const stringPayload = message.value.toString();
+        const jsonPayload = JSON.parse(stringPayload);
+
+        // Process only messages of our product
+        // Note: doing this before parsing to avoid errors on messages of other products
+        if (jsonPayload.productId !== productId) {
+          loggerInstance.info(
+            `Skipping message for partition ${partition} with offset ${message.offset} - Not required product: ${jsonPayload.productId}`
+          );
+          return;
+        }
+
+        if (jsonPayload.eventType !== "UPDATE") {
+          loggerInstance.info(
+            `Skipping message for partition ${partition} with offset ${message.offset} - Not required eventType: ${jsonPayload.eventType}`
+          );
+          return;
+        }
+
+        if (!allowedOriginsUuid.includes(jsonPayload.institutionId)) {
+          loggerInstance.warn(
+            `Skipping message for partition ${partition} with offset ${message.offset} - Not allowed origin.`
+          );
+          return;
+        }
+
+        // Note: not interested in users that are admin (they can be admin in a client) and that are active in that client
+        // We are interested in users that are not admin (lost admin role) or that are admin but not active in that client (lost tenant relationship)
+        if (
+          jsonPayload.user.productRole === userRole.ADMIN_ROLE &&
+          jsonPayload.user.relationshipStatus === "ACTIVE"
+        ) {
+          loggerInstance.info(
+            `Skipping message for partition ${partition} with offset ${message.offset} - User is a valid admin and relationshipStatus is ${jsonPayload.user.relationshipStatus}.`
+          );
+          return;
+        }
+
+        const token = (await refreshableToken.get()).serialized;
+        const headers = getInteropHeaders({ token, correlationId });
+
+        const response = await authorizationProcessClient.client.getClients({
+          queries: {
+            userIds: [jsonPayload.user.userId],
```

Comment:
The same as before, you don't have guarantee that `userId` is defined and have valid values.
But what happens if `userIds` is null or undefined ? Which is the response?
Also we need to using a needs to another query param here, we must understand if the clients have this user as Admin.
If not already implemented add TODO comment here and open a dedicated task on jira to track this task.

Reply:
As suggested by @AsterITA, this filter is no longer required. The value jsonPayload.user.userId defines the admin user ID and is probably not present in client users either.